lib/core/Observer.ts

```diff
@@ -102,8 +112,9 @@ export default class Observer {
         let transactions = readResult ? readResult.transactions : [];
         moreTransactions = readResult ? readResult.moreTransactions : false;
 
+        const transactionsToProcess = this.operationRateLimiter.getHighestFeeTransactionsPerBlock(transactions);
```

Comment:
Design: an alternative approach would be to make Observer aware of of different versions of rate limiting by:

1. (Optional first step) Ensuring the contract for fetching transactions from bitcoin service always returns complete set in same blockchain time/height.
2. Group the given transactions by their blockchain time/height.
3. Feed each transaction group to the correct version of rate limiter to select the qualified transactions.
4. Finally process the select transactions.

This logic probably involves 1 loop nested in another.